openedx/core/lib/api/tests/test_authentication.py

```diff
@@ -284,3 +291,40 @@ def test_responses_to_scoped_requests(self, scope_statuses):
         self.assertEqual(response.status_code, scope_statuses.read_status)
         response = self.post_with_bearer_token('/oauth2-with-scope-test/', token=self.access_token.token)
         self.assertEqual(response.status_code, scope_statuses.write_status)
+
+
+@ddt.ddt
+@unittest.skipUnless(settings.ROOT_URLCONF == 'lms.urls', 'Test only valid in lms')
+class TestJWTAuthentication(mixins.JwtMixin, TestCase):
+    """ Test id_token authentication."""
+
+    USERNAME = 'test-username'
+
+    def setUp(self):
+        self.user = UserFactory.create(username=self.USERNAME)
+        self.payload = self.default_payload(user=self.user)
+        self.jwt_token = self.generate_id_token(user=self.user)
+        self.authentication = JSONWebTokenAuthentication()
+        super(TestJWTAuthentication, self).setUp()
+
+    def _authenticate_credentials(self, payload):
+        """ Verify credentials by checking username."""
+        user = self.authentication.authenticate_credentials(payload)
+        self.assertEqual(user.username, self.USERNAME)
+
+    @patch.dict('django.conf.settings.FEATURES', {'ENABLE_JWT_AUTH': True})
+    def test_enabled_jwt_auth(self):
+        """ Ensure that the JWT auth works fine when its feature flag
+        'ENABLE_JWT_AUTH' is set.
+        """
+        self._authenticate_credentials(self.payload)
+        jwt_decode_handler(self.jwt_token)
+
+    @patch.dict('django.conf.settings.FEATURES', {'ENABLE_JWT_AUTH': False})
+    def test_disabled_jwt_auth(self):
+        """ Ensure that the JWT auth raises exception when its feature flag
+        'ENABLE_JWT_AUTH' is not set.
+        """
+        self._authenticate_credentials(self.payload)
```

Comment:
@zubair-arbi I'd suggest removing this step from the tests, and the setup that it uses, because it's not really adding anything to what's being tested.  Invoking the decode handler should be enough.